src/chainparams.cpp

```diff
@@ -501,21 +501,21 @@ void CRegTestParams::UpdateActivationParametersFromArgs(const ArgsManager& args)
         std::vector<std::string> vDeploymentParams;
         boost::split(vDeploymentParams, strDeployment, boost::is_any_of(":"));
         if (vDeploymentParams.size() != 3) {
-            throw std::runtime_error("Version bits parameters malformed, expecting deployment:start:end");
+            throw std::runtime_error("Version bits parameters malformed, expecting deployment:@startheight:@timeoutheight");
         }
-        int64_t nStartTime, nTimeout;
-        if (!ParseInt64(vDeploymentParams[1], &nStartTime)) {
-            throw std::runtime_error(strprintf("Invalid nStartTime (%s)", vDeploymentParams[1]));
+        int64_t startheight = 0, timeoutheight = 0;
+        if (vDeploymentParams[1].empty() || vDeploymentParams[1].front() != '@' || !ParseInt64(vDeploymentParams[1].substr(1), &startheight)) {
+            throw std::runtime_error(strprintf("Invalid startheight (%s)", vDeploymentParams[1]));
         }
-        if (!ParseInt64(vDeploymentParams[2], &nTimeout)) {
-            throw std::runtime_error(strprintf("Invalid nTimeout (%s)", vDeploymentParams[2]));
+        if (vDeploymentParams[2].empty() || vDeploymentParams[2].front() != '@' || !ParseInt64(vDeploymentParams[2].substr(1), &timeoutheight)) {
+            throw std::runtime_error(strprintf("Invalid timeoutheight (%s)", vDeploymentParams[2]));
```

Comment:
Could consider checking that `startheight` and `timeoutheight` (and `min_activation_height`) are either on retarget boundaries or match `-1, max int` or `-2, -2`. (Values not on retarget boundaries will be rounded up implicitly by the versionbits logic)

Reply:
Perhaps for a followup.

Reply:
I've decided to add checking for these parameters.